### PR TITLE
Grafana 2.1.0 and InfluxDB 0.9.4 Support E2E

### PR DESCRIPTION
**Pull request by thucatebay**
- Discover InfluxDB service endpoint and create a datasource using Grafana API.
- Import a couple of dashboards which use templating and repeated panels in Grafana 2.1.0.
- The dashboards pick up nodes, pods, and containers automatically with zero configurations.

### CODE REVIEW

**Comment by thucatebay** [comment]
@vishh I'd appreciate your comments on this PR. You can see the two dashboards that would get created out of the box at http://104.197.136.164/. This is my own Kubernetes cluster on GCE, with the changes in this PR. It works from end to end without manual configurations.

I built my own heapster image from master, which supports InfluxDB 0.9.4.

If this looks good I'll need your help with building kubernetes/grafana:2.1.0 from the Dockerfile, and also with doing another release of heapster from master.

Thanks.

**Comment by thucatebay** [comment]
I signed it! I'm contributing on behalf of eBay.

**Comment by vishh** [comment]
@thucatebay: This is great! We currently maintain all the dev side of monitoring in the [heapster](https://github.com/kubernetes/heapster) repo. Will you be able to move these changes to that repo? If not, I can help you with that.

File: cluster/addons/cluster-monitoring/influxdb/heapster-controller.yaml [hunk 1]
```diff
@@ -20,7 +20,7 @@ spec:
         kubernetes.io/cluster-service: "true"
     spec:
       containers:
-        - image: gcr.io/google_containers/heapster:v0.18.1
+        - image: thuctnguyen/heapster:v0.18.2-rc1
```
Comment (vishh) [inline cluster/addons/cluster-monitoring/influxdb/heapster-controller.yaml]: I assume this change was unintentional. Kindly revert these changes. In general, once we move to the main heapster repo, these changes will not be necessary.

**Comment by vishh** [comment]
One more meta comment: To support data retention while upgrading InfluxDB, we decided to upgrade to InfluxDB v0.8.9 before moving to v0.9.x. So Kubernetes v1.1 release will ship with influxDB v0.8.9. One of the subsequent patch releases will upgrade InfluxDB to v0.9.x.

**Comment by thucatebay** [comment]
@vishh thanks for the comments. I'll submit a PR in the heapster repository. I'll try to make it work with InfluxDB 0.8.9 for now in that PR. I'll do another one when we're ready to upgrade to 0.9.4.

**Comment by vishh** [comment]
So heapster HEAD now supports InfluxDB v0.9.4. The v0.18 release branch is
restricted to influxDB v0.8.9. This PR can go in as-is to HEAD. You might
have to tweak the PR to post a patch against the release-v0.18 branch
thought. Do take a look at the existing `grafana-dashboard.json` files in
the heapster repo. It should be under `docs`

File: cluster/addons/cluster-monitoring/influxdb/heapster-controller.yaml (hunk 1 above)
Comment (thucatebay) [inline cluster/addons/cluster-monitoring/influxdb/heapster-controller.yaml]: This PR depends on the latest heapster code with support for InfluxDB 0.9.4. I built heapster from HEAD and created thuctnguyen/heapster:v0.18.2-rc1 from that binary. Is there a Docker image for the latest heapster code that I can use?

Comment (vishh) [inline cluster/addons/cluster-monitoring/influxdb/heapster-controller.yaml]: Not yet. We can make an official heapster release with InfluxDB v0.9.x once
your PR lands.

**Comment by thucatebay** [comment]
Replaced by https://github.com/kubernetes/heapster/pull/599
